coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/app/config/L2NetworkGasPricingConfig.kt

```diff
@@ -0,0 +1,186 @@
+import net.consensys.linea.ethereum.gaspricing.staticcap.GasUsageRatioWeightedAverageFeesCalculator
+import net.consensys.linea.ethereum.gaspricing.staticcap.MinerExtraDataV1CalculatorImpl
+import net.consensys.linea.ethereum.gaspricing.staticcap.TransactionCostCalculator
+import net.consensys.linea.ethereum.gaspricing.staticcap.VariableFeesCalculator
+import net.consensys.zkevm.coordinator.app.L2NetworkGasPricingService
+import java.net.URL
+import java.time.Duration
+import kotlin.time.toKotlinDuration
+
+data class SampleTransactionGasPricingTomlDto(
+  val plainTransferCostMultiplier: Double = 1.0,
+  val compressedTxSize: Int = 125,
+  val expectedGas: Int = 21000
+)
+
+data class LegacyGasPricingTomlDto(
+  val type: Type,
+
+  val naiveGasPricing: NaiveGasPricingTomlDto?,
+  val sampleTransactionGasPricing: SampleTransactionGasPricingTomlDto = SampleTransactionGasPricingTomlDto(),
+  val gasPriceUpperBound: ULong,
+  val gasPriceLowerBound: ULong
+) {
+  enum class Type {
+    Naive,
+    SampleTransaction
+  }
+
+  init {
+    if (type == Type.Naive && naiveGasPricing == null) {
+      throw IllegalStateException("LegacyGasPricing $type configuration is null.")
+    }
+
+    require(gasPriceUpperBound >= gasPriceLowerBound) {
+      "gasPriceUpperBound must be greater than or equal to gasPriceLowerBound"
+    }
+  }
+}
+
+data class NaiveGasPricingTomlDto(
+  val baseFeeCoefficient: Double,
+  val priorityFeeCoefficient: Double,
+  val baseFeeBlobCoefficient: Double
+)
+
+data class VariableCostPricingTomlDto(
+  val gasPriceFixedCost: ULong,
+  val legacyFeesMultiplier: Double,
+  val margin: Double,
+  val variableCostUpperBound: ULong,
+  val variableCostLowerBound: ULong
+) {
+  init {
+    require(variableCostUpperBound >= variableCostLowerBound) {
+      "variableCostUpperBound must be greater than or equal to variableCostLowerBound"
+    }
+  }
+}
+
+data class JsonRpcPricingPropagationTomlDto(
+  override var disabled: Boolean = false,
+  val gethGasPriceUpdateRecipients: List<URL>,
+  val besuGasPriceUpdateRecipients: List<URL>
+) : FeatureToggleable {
+  init {
+    require(disabled || (gethGasPriceUpdateRecipients.isNotEmpty() || besuGasPriceUpdateRecipients.isNotEmpty())) {
+      "There is no point of enabling JSON RPC pricing propagation if there are no " +
+        "gethGasPriceUpdateRecipients or besuGasPriceUpdateRecipients defined"
+    }
+  }
+}
+
+data class ExtraDataPricingPropagationTomlDto(
+  override var disabled: Boolean = false,
+  val extraDataUpdateRecipient: URL
+) : FeatureToggleable
+
+data class L2NetworkGasPricingTomlDto(
+  override var disabled: Boolean = false,
+  override val requestRetry: RequestRetryConfigTomlFriendly,
+
+  val priceUpdateInterval: Duration,
+  val feeHistoryBlockCount: Int,
+  val feeHistoryRewardPercentile: Double,
+
+  val blobSubmissionExpectedExecutionGas: Int,
+  @ConfigAlias("bytesPerDataSubmission") val _bytesPerDataSubmission: Int?,
```

Comment:
what is config used for exactly?

Reply:
It's used to compute our data submission costs. In practice it matches l1BlobGas, which represent blob gas consumed per blob. But in case 1 byte of data doesn't cost 1 unit of blob gas, we'll have a way to change this.

It might be a far fetched example, but this is how it was implemented anyway